public/sw.js

```diff
@@ -92,11 +100,47 @@ self.addEventListener('fetch', (event) => {
 				}
 
 				// If the CSS/JS didn't change since it's been cached, return the cached version
-				if (hasHash(event.request.url) && hasSameHash(event.request.url, cached.url)) {
+				if (
+					hasHash(event.request.url) &&
+					hasSameHash(event.request.url, cached.url)
+				) {
 					return cached;
 				}
 			}
 			return fetchEvent;
 		}),
 	);
 });
+
+self.addEventListener('push', function (event) {
+	const data = JSON.parse(event.data.text());
+
+	const title = data.type === 'd' ? data.sender.name : data.name;
+	const message = data.type === 'd' ? data.message : data.sender.name + ": " + data.message;
+	let redirectURL = data.type === 'd' ? "/direct/" : data.type === 'c' ? "/channel/" : "/group/";
```

Comment:
This logic should be placed on server side, in notification should we just pass title, message, redirectURL, and keep all this logic at the place before sending push message from server side?